packages/xpetra/sup/Utils/Xpetra_MatrixUtils.hpp

```diff
@@ -434,13 +434,14 @@ class MatrixUtils {
     return bA;
   }
 
-  /** Given a matrix A, detect zero diagonals and replace any found with ones. */
+  /** Given a matrix A, detect too small diagonals and replace any found with ones. */
 
   static void CheckRepairMainDiagonal(RCP<Xpetra::Matrix<Scalar, LocalOrdinal, GlobalOrdinal, Node>>& Ac,
-                                 bool const &repairZeroDiagonals, Teuchos::FancyOStream &fos)
+                                 bool const &repairZeroDiagonals, Teuchos::FancyOStream &fos,
+                                 const typename Teuchos::ScalarTraits<Scalar>::magnitudeType threshold = Teuchos::ScalarTraits<double>::zero())
```

Comment:
Hm, why double, not Scalar?

Reply:
`threshold` is of type `magnitudeType`.   Would this work?  It looks terrible.
```
const typename Teuchos::ScalarTraits<Scalar>::magnitudeType threshold =
Teuchos::ScalarTraits<Scalar>::magnitude(Teuchos::ScalarTraits<Scalar>::zero())))

```

Reply:
I'm not sure it will. Alternatively, maybe try
```
Teuchos::ScalarTraits<typename Teuchos::ScalarTraits<Scalar>::magnitudeType>::zero()
```

Reply:
@cgcgcg 's snippet looks right as it matches the types of the left and right hand side of the assignment operator.

Reply:
Fixed with latest commit.